src/features/defi/providers/foxy/components/FoxyManager/Overview/Claim/ClaimStatus.tsx

```diff
@@ -100,12 +97,22 @@ export const ClaimStatus: React.FC<ClaimStatusProps> = ({ accountId }) => {
 
   const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
 
-  const accountFilter = useMemo(() => ({ accountId: accountId ?? '' }), [accountId])
-  const bip44Params = useAppSelector(state => selectBIP44ParamsByAccountId(state, accountFilter))
+  const dispatch = useAppDispatch()
+  // TODO: maybeRefetchOpportunities heuristics
```

Comment:
Not better, not worse than the current implementation, but still the wrong place for it

Reply:
agreed, let's figure out how the tx provider can register callbacks/listeners for all these different providers in a more elegant way